.github/workflows/auto-deploy-contracts.yaml

```diff
@@ -0,0 +1,114 @@
+name: Auto deploy IPC contracts when changed
+
+on:
+  workflow_dispatch:
+  # TODO: Test only. Remove before PR merge
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - contracts/**
+
+concurrency:
+  # Only allow one run at a time for this workflow
+  group: auto-deploy-contracts
+  cancel-in-progress: true
+
+jobs:
+  deploy-contracts:
+    runs-on: ubuntu-latest
+    env:
+      RPC_URL: https://calibration.filfox.info/rpc/v1
+      PRIVATE_KEY: ${{ secrets.CONTRACTS_DEPLOYER_PRIVATE_KEY }}
+    steps:
+      - name: Checkout cd/contract branch
+        uses: actions/checkout@v4
+        with:
+          ref: cd/contracts
+          submodules: recursive
+
+      - name: (Dry run) Try merge from main branch and update cd/contracts branch
```

Comment:
Dry run pull merge is to verify that the condition of cd/contracts branch is able to merge with latest main branch. If failed, we will resolve whatever error manually.

Reply:
Calling this a dry run is at bit confusing as its actually pulling and merging (so has side effects on the runner)... its just not pushing changes. 

Also "... and update cd/contracts branch" does not seem to be right

Reply:
Yes, the "dry" part mostly means that we will not modify remote code repos. This is useful because we can fail early when merging generate conflicts (rather than complain after deploying the contract).

I updated the step's name to better describe what it does.